src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,15 +199,21 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                // Following trick can reduce the range check by one comparison
+                if ((uint)index >= (uint)Count)
```

Comment:
Thanks, but there was no agreement in that issue on making such behavior changes... it was just one person's suggestion.